File: challenges/easy/divisor-of-a-number/solutions/aorlando/divisors-of-a-number.cpp
#include <iostream>
#include <vector>

using namespace std;


void divisorsOfNumber(int number) {
    int count = 0;
    vector<int> divisors;

    if(number == 0){
        cout << "Quitting... " << endl;
        exit(0);
    }

    for(int n=1; n <= number; n++) {

        if(number % n == 0) {
            divisors.push_back(n);
            count++;
        }
    }
    
    cout << number << " have " << count << " divisors:";
    for(int n=0; n <= divisors.size() - 1; n++) {
        cout << " " << divisors.at(n);
    }
    cout << endl;
}


int main () {

    int n, end;

    do {
        cout << "Enter a number (0 to exit): ";
        cin >> n;

        divisorsOfNumber(n);
    }
    while(end!=0);

    return 0;
}

File: challenges/easy/divisor-of-a-number/solutions/aorlando/divisors-of-a-number.py
def divisors_of_number(number):
    divisor = 0
    mytuple = tuple()
    for n in range(1, number+1):
        if number % n == 0:
            divisor += 1
            mytuple += (n,)
    return f"{number} have {divisor} divisors => {mytuple}"


test1 = divisors_of_number(36)
test2 = divisors_of_number(47)
test3 = divisors_of_number(3)
test4 = divisors_of_number(1)
test5 = divisors_of_number(276)

print(test1)
print(test2)
print(test3)
print(test4)
print(test5)
